### DISCUSSION TITLE
Question about OpenFAST: How to use parallel computing.

### DISCUSSION THREAD

**Discussion by niay000** [OpenFAST/openfast#1865] [Q&A]
Dear all,

I’m using OpenFAST v3.0.0. I would like to ask if OpenFast can use parallel computing, I want to shorten the calculation time because I do a lot of calculations.
I read the documentation of OpenFAST. And I tried the following methods.
But the calculation time doesn't seem to be reduced.
Any help would be appreciated,

Best regards,

~~~
2.2.5. CMake with Make for Linux/macOS
~~~
Compile in parallel by adding “-jN” to the command where N is the number of makeparallel processes to use; i.e. .make -j4 openfast

**Comment by jjonkman** [comment]
Dear @niay000,

The only parallelization currently supported in OpenFAST is OpenMP parallelization that can speed up simulations with OLAF and FAST.Farm enabled.  But if you are not using OLAF or FAST.Farm, this parallelization wont help you.

We presented a new tight coupling algorithm for OpenFAST a week ago at the NAWEA / WindTech 2023 conference that offers a 10x-100x speed up (depending on the case) for models that previously had structure-structure coupling between BeamDyn-ElastoDyn and/or ElastoDyn-SubDyn.  The speed up tight coupling brings comes from the ability to use larger time steps rather than through parallelization of the algorithm.  Though not officially released yet, you can track the progress here: https://github.com/OpenFAST/openfast/pull/1850.

Best regards,

**Reply by niay000** [reply]
Dear jjonkman,

Thank you for your reply.
I appreciate your prompt response. It was very helpful.
I will review the documents at my earliest convenience.

Best regards,
